main.go

```diff
@@ -164,15 +180,23 @@ func main() {
 
 	// Configure the metric resources
 	if err := metricsConfig.Configure(ctx); err != nil {
-		log.Error(err, "error setting up metrics")
+		setupLog.Error(err, "error setting up metrics")
+		os.Exit(1)
+	}
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
```

Comment:
Isn't this a new introduction? We should call this out in the commit message. Ideally this would be in a follow up PR. This PR should not change any behavior.

Reply:
This was generated code for new controllers. I can take it out if thats ideal

Reply:
I think just adding this here is not sufficient. Don't you also need `livenessProbe` and `readinessProbe` entries in the `Deployment`?